### PR TITLE
Adjust the code to work on MIMIC dataset

### PR DESCRIPTION
**Pull request by Mohammeda4t1qbit**

### CODE REVIEW

File: constants/constants.py [hunk 1]
```diff
@@ -10,10 +10,18 @@
...
+              "Atelectasis",
+              "Consolidation",
+              "Emphysema",
+              "Linear Scar or Fibrotatic Bands",
+              "Lung Cavity",
+              "Lung Cyst",
+              "Nodule Mass",
+              "Pleural Effusion",
+              "Pleural Thickening",
+              "Pulmonary Edema",
```
Comment (emranmdanas) [inline constants/constants.py]: Replace Pulmonary Edema by Interstitial fluid, it was suggested by D

File: constants/constants.py [hunk 2]
```diff
@@ -10,10 +10,18 @@
...
+              "Undifferentiated Lung Opacity"]
```
Comment (emranmdanas) [inline constants/constants.py:26]: Pneumothorax is missing

File: mimic_convert_to_csv.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            report = report[examination_match.span()[1]:]
+
+    report = re.sub(r"\s{2,}", " ", report)
+    report = re.sub("$\n", "", report)
+    return report
+
+
+NLP_csv = pd.read_csv(nlp_csv_file, usecols=['path', 'report', 'ml_predictions'])
+NLP_csv = NLP_csv.rename(columns={"report": "org_report"})
+NLP_csv["report"] = None
```
Comment (emranmdanas) [inline mimic_convert_to_csv.py]: we can replace lines 34-40
NLP_csv["report"] = NLP_csv.apply(lambda row: clean_mimic(row["org_report"]), axis=1)
It should be few times faster than using loop.

**Review by emranmdanas — CHANGES_REQUESTED** [review]
Hi Mohammed, looks OK to me. Thanks for working with the repo. I've few comments. Please also update the phrases from the shared xls file.

File: mimic_convert_to_csv.py (hunk 3 above)
Comment (EhsanI4t1qbit) [reply mimic_convert_to_csv.py]: I think the following is also working in here!
`NLP_csv["report"] = NLP_csv["org_report"].apply(lambda x: clean_mimic(x))`

File: label.py [hunk 4]
```diff
@@ -36,16 +42,38 @@ def label(args):
...
+
+    """Load and clean the reports."""
+    df = pd.read_csv(parser.parse_args().reports_path)
+    max_workers = multiprocessing.cpu_count()
+    print(f"Generate labels using {max_workers} CPUs")
+    df_split = np.array_split(df, max_workers)
+
+    with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+        list(tqdm(executor.map(label, repeat(parser.parse_args()), [df_split[i] for i in range(max_workers)],
+                               [i for i in range(max_workers)]), total= max_workers))
```
Comment (EhsanI4t1qbit) [inline label.py:71]: Does replacing `[df_split[i] for i in range(max_workers)]` with just `df_split` also work?

File: constants/constants.py (hunk 1 above)
Comment (Mohammeda4t1qbit) [reply constants/constants.py]: Done

File: constants/constants.py (hunk 2 above)
Comment (Mohammeda4t1qbit) [reply constants/constants.py:26]: Done

File: mimic_convert_to_csv.py (hunk 3 above)
Comment (Mohammeda4t1qbit) [reply mimic_convert_to_csv.py]: Done , it is super fast now

File: label.py (hunk 4 above)
Comment (Mohammeda4t1qbit) [reply label.py:71]: df_split is a list of small dataframes , so we need to use df_split[i]

**Review by EhsanI4t1qbit — APPROVED** [review]
Good job Mohammed!

File: phrases/mention/consolidation.txt [hunk 5]
```diff
@@ -1 +1,26 @@
-consolidat
\ No newline at end of file
+consolidat
+airspace density
+airspace densities
+air bronchograms
+confluent airspace disease
+focal consolidation
+pneumonia
+pna
```
Comment (emranmdanas) [inline phrases/mention/consolidation.txt]: pulmonary nodular amyloidosis

File: phrases/unmention/nodule_mass.txt [hunk 6]
```diff
@@ -0,0 +1,4 @@
+massive
```
Comment (emranmdanas) [inline phrases/unmention/nodule_mass.txt:1]: massive
massengale
cavitated nodul
lymphangitic carcinomatosis
pleural based mass
pleural mass

File: phrases/mention/pleural_effusion.txt [hunk 7]
```diff
@@ -1,2 +1,17 @@
 pleural fluid
-effusion
\ No newline at end of file
+effusion
```
Comment (emranmdanas) [inline phrases/mention/pleural_effusion.txt:2]: unmention for pleural_effusion is missing

**Review by emranmdanas — CHANGES_REQUESTED** [review]
Few comments on phrases.

File: phrases/mention/consolidation.txt (hunk 5 above)
Comment (emranmdanas) [reply phrases/mention/consolidation.txt:8]: pna --> pulmonary nodular amyloidosis

File: phrases/unmention/nodule_mass.txt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+massengale
```
Comment (emranmdanas) [inline phrases/unmention/nodule_mass.txt:2]: please check https://docs.google.com/spreadsheets/d/1HiAP8jnbDw3uiMhNLSou2S_4lOSvJoKbpmDfKZVBXUo/edit#gid=2140168161

You;re missing one phrase

File: phrases/mention/consolidation.txt (hunk 5 above)
Comment (emranmdanas) [reply phrases/mention/consolidation.txt]: pna --> pulmonary nodular amyloidosis

**Review by emranmdanas — APPROVED** [review]

**Pull request merged**